engine/schema/src/com/cloud/host/dao/HostDaoImpl.java

```diff
@@ -1087,4 +1087,11 @@ public HostVO findByPublicIp(String publicIp) {
         sc.addAnd("dataCenterId", SearchCriteria.Op.EQ, zoneId);
         return customSearch(sc, null);
     }
+
+    @Override
+    public List<HostVO> listByType(Host.Type type) {
+        SearchCriteria<HostVO> sc = TypeSearch.create();
+        sc.setParameters("type", type);
+        return listBy(sc);
```

Comment:
the listByType consumers are not doing != null checks, we can fix it by doing it here and returning a Collections.emptyList(); this way the methods ensures that no null is returned

Please fix the NPE case, otherwise LGTM